core/ui/src/main/java/com/guidal/core/ui/components/HomeNavigationButton.kt

```diff
@@ -0,0 +1,114 @@
+package com.guidal.core.ui.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.interaction.collectIsPressedAsState
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.guidal.core.ui.models.UiModelMenuButtonIcon
+import com.guidal.core.ui.theme.GuidalIcons
+import com.guidal.core.ui.theme.GuidalTheme
+
+@Composable
+fun HomeNavigationButton(
+    label: String,
+    type: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    sectionIcon: UiModelMenuButtonIcon? = null,
```

Comment:
Why `sectionIcon` is optional? Is there a scenario where you won't use icon? As far as I am concerned the navigation by category would always have icon.

Reply:
Sure, agree